src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -789,9 +790,25 @@ private int updateStopTimesForPatternStops(List<PatternStop> patternStops) throw
         for (String tripId : timesForTripIds.keySet()) {
             // Initialize travel time with previous stop time value.
             int cumulativeTravelTime = timesForTripIds.get(tripId);
+            int timepointNumber = 0;
+            double timepointSpeed = 0;
+            double previousShapeDistTraveled = 0;
+            PatternStop currentTimepoint = patternStops.get(0); // First stop must be a timepoint
             for (PatternStop patternStop : patternStops) {
+                boolean isTimepoint = patternStop.timepoint == 1;
+                // If we have reached a timepoint (which is not the last), we calculate the speed between it and the next timepoint.
+                if (isTimepoint && interpolateStopTimes && timepointNumber < timepoints.size()-1){
+                    timepointNumber++;
```

Comment:
Include the `++` in the `if`

Reply:
I was gonna do this, but was worried it made the code less readable.

Reply:
Updated this in https://github.com/conveyal/gtfs-lib/pull/399/commits/b585c9df6e944b927859e7f6948dc5c2ab1080ff